tritium-caffeine/src/main/java/com/palantir/tritium/metrics/caffeine/CacheStats.java

```diff
@@ -70,6 +71,28 @@ public static CacheStats of(TaggedMetricRegistry taggedMetricRegistry, @Safe Str
         return new CacheStats(CacheMetrics.of(taggedMetricRegistry), name);
     }
 
+    /**
+     * Registers additional metrics for a Caffeine cache.
+     *
+     * @param cache cache for which to register metrics
+     */
+    public <K, V> void register(Cache<K, V> cache) {
+        checkState(
+                cache.policy().isRecordingStats(),
+                "Registered cache is not recording stats. Registered caches must enabled stats recording with "
+                        + ".recordStats(stats).");
+
+        metrics.estimatedSize().cache(name).build(cache::estimatedSize);
+        metrics.weightedSize().cache(name).build(() -> cache.policy()
+                .eviction()
+                .flatMap(e -> e.weightedSize().stream().boxed().findFirst())
+                .orElse(null));
+        metrics.maximumSize().cache(name).build(() -> cache.policy()
+                .eviction()
+                .map(Policy.Eviction::getMaximum)
+                .orElse(null));
+    }
```

Comment:
This is basically the API before https://github.com/palantir/tritium/pull/1920.

I decided to make this public so consumers have the option to register caches without the somewhat awkward (but less error-prone) fluent API.